### DISCUSSION TITLE
Can I use closed source code in my repo?

### DISCUSSION THREAD

**Discussion by cfpwastaken** [community/community#21662] [New to GitHub]
<p>So I have this problem.<br>
I wanna post code from a closed-source project that I modified.<br>
I don’t know if I’m allowed to do this since its closed source and that definitely for reasons (that I actually don’t know)<br>
So how can I do this? Am I allowed to do it? Can I just straight up NOT do it?<br>
Greetings</p>

**Comment by airtower-luna** [comment]
<p>That sounds like the real question is: Would it be a copyright violation if you post that code? From your question I’m kind of guessing it would be, but for a reliable answer to that question I’m afraid you’ll need to ask a lawyer who’s familiar with copyright law.</p>

**Comment by cfpwastaken** [comment]
<p>Okay. It seems like I can’t and won’t post it because I clearly don’t want any copyright problems.<br>
It’s Minecraft by the way.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Mind that I’m only guessing, “code from a closed-source project” always smells like trouble unless you own it or have explicit permission. :wink:</p>
<p>But there are a lot of details to consider (e.g. the exact license terms, what exactly you’re doing with the code, copyright exceptions, …). If you’re not sure after reading the relevant license(s) what you need is legal advice, and I’m afraid a forum like this isn’t the right place for that. :sweat_smile:</p>
